core/src/tangram.cpp

```diff
@@ -109,7 +109,10 @@ Map::~Map() {
     // The unique_ptr to Impl will be automatically destroyed when Map is destroyed.
     impl->tileWorker.stop();
     impl->asyncWorker.reset();
-    impl->jobQueue.runJobs();
+
+    // Make sure other threads are stopped before calling stop()!
+    // All jobs will be executed immediately on add() afterwards.
+    impl->jobQueue.stop();
```

Comment:
Yeah, the destruction process relies on the ordering of members right now which isn't ideal.

The case of holding onto tasks that invoke disposers does seem pretty risky, this change sounds like a good idea.